packages/lib-classifier/src/hooks/useSessionStorage.js

```diff
@@ -0,0 +1,25 @@
+import { addDisposer, onSnapshot } from 'mobx-state-tree'
+import { useState } from 'react'
+
+import { asyncSessionStorage } from '@helpers'
+
+function persist(storageKey, store, storage) {
+  function _saveSnapshot(snapshot) {
+    const data = JSON.stringify(snapshot)
+    storage.setItem(storageKey, data)
+  }
+  const snapshotDisposer = onSnapshot(store, _saveSnapshot)
+  return addDisposer(store, snapshotDisposer)
+}
+
+export default function useSessionStorage(enableStorage, store, storageKey) {
+  const [loaded, setLoaded] = useState(false)
```

Comment:
This function is designed to stop running when this flips from `false` to `true`.